### PR TITLE
[FLINK-16455][hive] Introduce flink-sql-connector-hive modules to provide hive uber jars

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

Provide uber jars to improve hive dependencies out-of-box experience.

## Brief change log

- Introduce flink-sql-connector-hive-1.2.2
- Introduce flink-sql-connector-hive-2.2.0
- Introduce flink-sql-connector-hive-2.3.6
- Introduce flink-sql-connector-hive-3.1.2

## Verifying this change

Manual testing

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/hive/index.md [hunk 1]
```diff
@@ -92,6 +92,34 @@ to make the integration work in Table API program or SQL in SQL Client.
 Alternatively, you can put these dependencies in a dedicated folder, and add them to classpath with the `-C`
 or `-l` option for Table API program or SQL Client respectively.
 
+Apache Hive is built on Hadoop, so you need Hadoop dependency first, please refer to
+[Providing Hadoop classes]({{ site.baseurl }}/ops/deployment/hadoop.html#providing-hadoop-classes).
+
+There are two way to hive dependencies. First way, using bundled hive jar, providing multiple bundle
+hive jars that can cover all remote metastore versions. Second way, user defined dependencies, you
+can build your own dependencies if you need to.
```
Comment (lirui-apache) [inline docs/dev/table/hive/index.md]:
```suggestion
There are two ways to add Hive dependencies. First is to use Flink's bundled Hive jars. You can choose a bundled Hive jar according to the version of the metastore you use. Second is to add each of the required jars separately. The second way can be useful if the Hive version you're using is not listed here.
```

File: docs/dev/table/hive/index.md [hunk 2]
```diff
@@ -92,6 +92,34 @@ to make the integration work in Table API program or SQL in SQL Client.
...
+hive jars that can cover all remote metastore versions. Second way, user defined dependencies, you
+can build your own dependencies if you need to. 
+
+#### Using bundled hive jar
+
+The following tables list all available bundled hive jars. You can pick one to the `/lib/` directory in Flink distribution.
+
+{% if site.is_stable %}
+
+| Remote Metastore   | Dependency Version  | Maven dependency             | SQL Client JAR         |
```
Comment (lirui-apache) [inline docs/dev/table/hive/index.md]: I don't think the `Dependency Version` column is useful. We just need to let users know which jar to download (or which mvn dependency to use), given the version of their HMS.

File: docs/dev/table/hive/index.md [hunk 3]
```diff
@@ -281,6 +261,11 @@ Please find the required dependencies for different Hive major versions below.
 </div>
 </div>
 
+If you use the hive version of HDP or CDH, you need to refer to the dependency in the previous section and select a similar version.
+
+And you need to specify selected and supported "hive-version" in yaml, HiveCatalog and HiveModule.
```
Comment (lirui-apache) [inline docs/dev/table/hive/index.md]: I think providing an example for this would be more helpful

Comment (JingsongLi) [reply docs/dev/table/hive/index.md]: I don't want provide too detailed information. This is the home page of hive. These words should be enough.

**Comment by JingsongLi** [comment]
@flinkbot run travis

File: flink-connectors/flink-sql-connector-hive-1.2.2/pom.xml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-hive_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>*</groupId>
+					<artifactId>*</artifactId>
```
Comment (bowenli86) [inline flink-connectors/flink-sql-connector-hive-1.2.2/pom.xml]: what does this part mean? exclude all the dependencies?

Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-1.2.2/pom.xml]: Yes, we only use explicit dependencies.

**Comment by JingsongLi** [comment]
Thanks @bowenli86 and @lirui-apache for your review. Merging...

**Comment by pnowojski** [comment]
I believe
> Dependencies (does it add or upgrade a dependency): (yes / ** -> no <-**)

is not correct

**Pull request merged**